server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -286,6 +286,36 @@ public final QueryBuilder rewrite(QueryRewriteContext queryRewriteContext) throw
     }
 
     protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws IOException {
+        if (queryRewriteContext == null) {
```

Comment:
There is at least case when this can be null and it is uncovered by a test that deals with fuzzy queries.

```
./gradlew ':rest-api-spec:yamlRestTest' --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT.test {yaml=search/320_disallow_queries/Test disallow expensive queries}" -Dtests.seed=C6EE7229D969ECFD -Dtests.locale=es-SV -Dtests.timezone=Asia/Novosibirsk -Druntime.java=20
```

I will try to debug it and see if I can understand why it is null in such a scenario.

Reply:
Unfortunately the dafult implementation of `convertToSearchExecutionContext` cannot return `this` because a `SearchExecutionContext` is expected instead of a `QueryRewriteContext` as a return value...so we need to return null there.

Reply:
ok, maybe add a comment on why this is needed? Seems like it's a question we may ask ourselves again soon :)